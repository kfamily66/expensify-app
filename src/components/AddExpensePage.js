import React from "react";

const AddExpensePage = () => <div>This is a create page!</div>;

export default AddExpensePage;
