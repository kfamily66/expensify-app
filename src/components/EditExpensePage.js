import React from "react";

const EditExpensePage = props => {
  console.log(props);
  return <div>This is an edit page!</div>;
};

export default EditExpensePage;
