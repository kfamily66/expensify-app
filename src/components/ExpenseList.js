import React from "react";
import { connect } from "react-redux";

const ExpenseList = props => (
  <div>
    <h2>Expense list</h2>
    {props.expenses.length}
    {props.filters.text}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: state.expenses,
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseList);
