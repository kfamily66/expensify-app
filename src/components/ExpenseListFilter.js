import React from "react";
import { connect } from "react-redux";
import "react-dates/initialize";
import { DateRangePicker } from "react-dates";
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from "../actions/filters";

export class ExpenseListFilter extends React.Component {
  state = {
    focused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = focusedInput => {
    this.setState(() => ({
      focused: focusedInput
    }));
  };
  onTextChange = e => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = e => {
    if (e.target.value === "date") {
      this.props.sortByDate();
    } else if (e.target.value === "amount") {
      this.props.sortByAmount();
    }
  };
  render() {
    return (
      <div>
        <input type="text" value={this.props.filters.text} onChange={this.onTextChange} />
        <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId={"id1"}
          endDate={this.props.filters.endDate}
          endDateId={"id2"}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.focused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          firstDayOfWeek={1}
          transitionDuration={100}
          displayFormat={"DD/MM/YYYY"}
          hideKeyboardShortcutsPanel={true}
          showClearDates={true}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = dispatch => ({
  setStartDate: startDate => dispatch(setStartDate(startDate)),
  setEndDate: endDate => dispatch(setEndDate(endDate)),
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ExpenseListFilter);
