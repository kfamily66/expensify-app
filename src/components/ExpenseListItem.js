import React from "react";
import { Link } from "react-router-dom";

export default ({ description, amount, createdAt, id }) => (
  <div>
    <h3>
      <Link to={`/edit/${id}`}>{description}</Link>
    </h3>
    <p>
      {amount} - {createdAt}
    </p>
  </div>
);
