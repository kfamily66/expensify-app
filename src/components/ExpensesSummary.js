import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const formattedTotal = numeral(expensesTotal / 100).format("$0,0.00");
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";

  return (
    <div>
      <h3>
        Viewing {expenseCount} {expenseWord} totalling {formattedTotal}
      </h3>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
