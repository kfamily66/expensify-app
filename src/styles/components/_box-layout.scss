.box-layout {
  background: url("/images/back1.png");
  background-size: cover;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

.box-layout__box {
  background: fade-out(white, 0.35);
  border-radius: 3px;
  padding: $l-size $m-size;
  text-align: center;
  width: 25rem;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}
