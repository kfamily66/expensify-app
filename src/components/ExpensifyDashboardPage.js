import React from "react";
import ExpenseList from "./ExpenseList";

const ExpensifyDashboardPage = () => (
  <div>
    <ExpenseList />
  </div>
);

export default ExpensifyDashboardPage;
