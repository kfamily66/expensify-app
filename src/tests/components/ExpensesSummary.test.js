import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should render ExpensesSummary without visible expenses", () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={1} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSummary without visible expenses", () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={550} />);
  expect(wrapper).toMatchSnapshot();
});
