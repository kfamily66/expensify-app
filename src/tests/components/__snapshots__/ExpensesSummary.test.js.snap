// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with visible expenses 1`] = `
<div>
  <h3>
    Viewing 
    3
     
    expenses
     totalling 
    $5.50
  </h3>
</div>
`;

exports[`should render ExpensesSummary without visible expenses 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $0.01
  </h3>
</div>
`;
